lib/Onyx.js

```diff
@@ -359,18 +365,20 @@ function connect(mapping) {
         return connectionID;
     }
 
-    // Check to see if this key is flagged as a safe eviction key and add it to the recentlyAccessedKeys list
-    if (mapping.withOnyxInstance && !isCollectionKey(mapping.key) && isSafeEvictionKey(mapping.key)) {
+    // Commit connection only after init passes
+    deferredInit.promise.then(() => {
```

Comment:
This promise is resolved after Onyx.init runs
Moved the original logic inside the `then` due to the `isSafeEvictionKey` check - it seems those keys are set during `Onyx.init` and this check would not work correctly before `Onyx.init` is complete

Reply:
Let's put this `.then()` on a new line if we're going to chain  ?